app/uk/gov/hmrc/helptosave/services/HelpToSaveService.scala

```diff
@@ -133,58 +137,49 @@ class HelpToSaveServiceImpl @Inject()(
         }
     })
 
-  def getPersonalDetails(nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext): Result[PayePersonalDetails] =
-    EitherT({
-      val timerContext = metrics.payePersonalDetailsTimer.time()
-
-      dESConnector
-        .getPersonalDetails(nino)
-        .map[Either[String, PayePersonalDetails]] {
-          response =>
-            val time = timerContext.stop()
-
-            val additionalParams = "DesCorrelationId" -> response.desCorrelationId
-
-            response.status match {
-              case Status.OK =>
-                val result = response.parseJsonWithoutLoggingBody[PayePersonalDetails]
-                result.fold(
-                  { e =>
-                    metrics.payePersonalDetailsErrorCounter.inc()
-                    logger.warn(
-                      s"Could not parse JSON response from paye-personal-details, received 200 (OK): $e ${timeString(time)}",
-                      nino,
-                      additionalParams)
-                    pagerDutyAlerting.alert("Could not parse JSON in the paye-personal-details response")
-                  },
-                  _ =>
-                    logger.debug(
-                      s"Call to check paye-personal-details successful, received 200 (OK) ${timeString(time)}",
-                      nino,
-                      additionalParams)
-                )
-                result
-
-              case other =>
-                logger.warn(
-                  s"Call to paye-personal-details unsuccessful. Received unexpected status $other ${timeString(time)}",
-                  nino,
-                  additionalParams)
+  def getPersonalDetails(nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext): Result[PayePersonalDetails] = {
+    val ifSwitch: Boolean = appConfig.ifEnabled
+    val (response, tag) =
+      if (ifSwitch) {
+        (iFConnector.getPersonalDetails(nino), "[IF]")
+      } else {
+        (dESConnector.getPersonalDetails(nino), "[DES]")
+      }
+    val timerContext = metrics.payePersonalDetailsTimer.time()
+    response.map { response =>
+        val time = timerContext.stop()
+        val params = (response: HttpResponse) => "CorrelationId" -> response.correlationId
+      response.status match {
+          case Status.OK =>
+            response.parseJsonWithoutLoggingBody[PayePersonalDetails] tap {
+              case Left(e) =>
                 metrics.payePersonalDetailsErrorCounter.inc()
-                pagerDutyAlerting.alert("Received unexpected http status in response to paye-personal-details")
-                Left(s"Received unexpected status $other")
-
+                logger.warn(
+                  s"$tag Could not parse JSON response from paye-personal-details, received 200 (OK): $e ${timeString(time)}",
```

Comment:
what was the verdict on the tag? is it still needed?

Reply:
It's not needed in alert, I hope we can keep them in warn and debug levels